catalog/plugin/catalog-plugin-metacardingest-network/src/main/java/org/codice/ddf/catalog/plugin/metacard/MetacardIngestNetworkPlugin.java

```diff
@@ -0,0 +1,174 @@
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.plugin.metacard;
+
+import static org.apache.commons.lang.BooleanUtils.isFalse;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.shiro.util.ThreadContext;
+import org.codice.ddf.catalog.plugin.metacard.util.AttributeFactory;
+import org.codice.ddf.catalog.plugin.metacard.util.KeyValueParser;
+import org.codice.ddf.catalog.plugin.metacard.util.MetacardServices;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.DeleteResponse;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PreAuthorizationPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+
+/**
+ * Conditional attribute adjustments for metacards based on IP, hostname, scheme, and context path.
+ */
+public class MetacardIngestNetworkPlugin implements PreAuthorizationPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MetacardIngestNetworkPlugin.class);
+
+    private static final String CLIENT_INFO_KEY = "client-info";
+
+    private final KeyValueParser keyValueParser;
+
+    private final MetacardServices metacardServices;
+
+    private final AttributeFactory attributeFactory;
+
+    private final MetacardCondition metacardCondition;
+
+    private List<String> newAttributes;
+
+    /**
+     * Default constructor.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            MetacardServices metacardServices) {
+        this(keyValueParser, metacardServices, new AttributeFactory(), new MetacardCondition());
+    }
+
+    /**
+     * Extended constructor with an {@link AttributeFactory} and {@link MetacardCondition} to provide
+     * more fine-grained control.
+     */
+    public MetacardIngestNetworkPlugin(KeyValueParser keyValueParser,
+            MetacardServices metacardServices, AttributeFactory attributeFactory,
+            MetacardCondition metacardCondition) {
+        this.keyValueParser = keyValueParser;
+        this.metacardServices = metacardServices;
+        this.attributeFactory = attributeFactory;
+        this.metacardCondition = metacardCondition;
+    }
+
+    public String getCriteriaKey() {
+        return metacardCondition.getCriteriaKey();
+    }
+
+    public void setCriteriaKey(String criteriaKey) {
```

Comment:
Ok. Legitimate threading issue. Going to move toward the `component-managed` strategy and make the `MetacardCondition` immutable.